src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1437,16 +1437,16 @@ namespace
 void addTableDependencies(const ASTCreateQuery & create, const ASTPtr & query_ptr, const ContextPtr & context)
 {
     QualifiedTableName qualified_name{create.getDatabase(), create.getTable()};
-    auto ref_dependencies = getDependenciesFromCreateQuery(context->getGlobalContext(), qualified_name, query_ptr, context->getCurrentDatabase());
-    auto loading_dependencies = getLoadingDependenciesFromCreateQuery(context->getGlobalContext(), qualified_name, query_ptr);
+    auto ref_dependencies = getDependenciesFromCreateQuery(context->getGlobalContext(), qualified_name, query_ptr, context->getCurrentDatabase(), /*can_throw*/ true);
```

Comment:
Looks like we call `addTableDependencies` after calling doCreateTable (which add the new table/dictionary to the database, and contains something like a "commit point"): https://github.com/ClickHouse/ClickHouse/blob/e7e30d67fb2e2fdd603498f04f2aed5f11d48e0f/src/Interpreters/InterpreterCreateQuery.cpp#L1694-L1701

So it's too late to check if the query is valid or not, the table is already created at this moment

Reply:
Yes, I know, but I though that it still can be useful to make it unified with other calls here (perhaps some other check will throw some error at that point). Having said that, I don't have a really strong opinion here, so I can remove it from here if you think it makes more sense

Reply:
I don't think we should pass can_throw=true here, it makes more sense in `checkTableCanBeAddedWithNoCyclicDependencies`